source/d3d12/runtime_d3d12.cpp

```diff
@@ -1400,4 +1453,293 @@ void reshade::d3d12::runtime_d3d12::render_imgui_draw_data(ImDrawData *draw_data
+			{
+				const auto texture_data = tex.impl->as<d3d12_tex_data>();
+
+				if (texture_data->descriptors == nullptr)
+				{
+					{
+						D3D12_DESCRIPTOR_HEAP_DESC heap_desc = { D3D12_DESCRIPTOR_HEAP_TYPE_CBV_SRV_UAV };
+						heap_desc.NumDescriptors = 1;
+						heap_desc.Flags = D3D12_DESCRIPTOR_HEAP_FLAG_SHADER_VISIBLE;
+
+						if (FAILED(_device->CreateDescriptorHeap(&heap_desc, IID_PPV_ARGS(&texture_data->descriptors))))
+							return false;
+					}
+				}
+
+				D3D12_CPU_DESCRIPTOR_HANDLE srv_cpu_handle = texture_data->descriptors->GetCPUDescriptorHandleForHeapStart();
+
+				for (const auto &technique : _techniques)
+				{
+					if (_effect_data.size() >= technique.effect_index)
+						continue;
+
+					d3d12_effect_data &effect_data = _effect_data[technique.effect_index];
+
+					tex.width = frame_width();
+					tex.height = frame_height();
+
+					texture_data->resource.reset();
+					texture_data->resource = _depthstencil_texture;
+
+					{   D3D12_SHADER_RESOURCE_VIEW_DESC desc = {};
+					desc.Format = make_dxgi_format_normal(texture_data->resource->GetDesc().Format);
+					desc.ViewDimension = D3D12_SRV_DIMENSION_TEXTURE2D;
+					desc.Shader4ComponentMapping = D3D12_DEFAULT_SHADER_4_COMPONENT_MAPPING;
+					desc.Texture2D.MipLevels = 1;
+
+					// D3D12_CPU_DESCRIPTOR_HANDLE srv_handle = effect_data.srv_cpu_base;
+					D3D12_CPU_DESCRIPTOR_HANDLE srv_cpu_handle = texture_data->descriptors->GetCPUDescriptorHandleForHeapStart();
+					// srv_cpu_handle.ptr += _srv_handle_size;
+
+					_device->CreateShaderResourceView(texture_data->resource.get(), &desc, srv_cpu_handle);
+					}
+				}
+			}
+		}
+	}
+
+	return true;
+}
+
+com_ptr<ID3D12Resource> reshade::d3d12::runtime_d3d12::select_depth_texture_save(D3D12_RESOURCE_DESC &textureDesc, const D3D12_HEAP_PROPERTIES *props, D3D12_HEAP_FLAGS *heapFlags)
+{
+	// Function that selects the appropriate texture where we want to save the depth texture before it is cleared
+	// If this texture is null, create it according to the dimensions and the format of the depth texture
+	// Doing so, we avoid to create a new texture each time the depth texture is saved
+
+	textureDesc.Format = make_dxgi_format_typeless(textureDesc.Format);
+
+	// Create an unique index based on the texture format and dimensions
+	UINT idx = textureDesc.Format * textureDesc.Width * textureDesc.Height;
+
+	if (const auto it = _depth_texture_saves.find(idx); it != _depth_texture_saves.end())
+		return it->second;
+
+	if(!(textureDesc.Flags & D3D12_RESOURCE_FLAG_ALLOW_DEPTH_STENCIL))
+		textureDesc.Flags |= D3D12_RESOURCE_FLAG_ALLOW_DEPTH_STENCIL;
+
+	// Create the saved texture pointed by the index if it does not already exist
+	com_ptr<ID3D12Resource> depth_texture_save;
+
+	// Render targets are always either cleared to zero or not cleared at all (see 'ClearRenderTargets' pass state), so we can set the optimized clear value here to zero
```

Comment:
This comment doesn't make sense in this context =P

Reply:
Yes. Copy/paste stuff. Can be removed